jenkins_job_wrecker/modules/publishers.py

```diff
@@ -75,33 +75,179 @@ def fingerprinter(top, parent):
     parent.append({'fingerprint': fingerprint})
 
 
+def __process_email(trigger, ext_email):
+    for email in trigger:
+        if email.tag == "email":
+            send_to = []
+            for email_child in email:
+                if email_child.tag == "recipientProviders":
+                    for provider in email_child:
+                        if provider.tag == \
+                                "hudson.plugins.emailext.plugins.recipients.UpstreamComitterRecipientProvider":
+                            send_to.append("upstream-committers")
+                        elif provider.tag == \
+                                "hudson.plugins.emailext.plugins.recipients.DevelopersRecipientProvider":
+                            send_to.append("developers")
+                        elif provider.tag == \
+                                "hudson.plugins.emailext.plugins.recipients.RequesterRecipientProvider":
+                            send_to.append("requester")
+                        elif provider.tag == \
+                                "hudson.plugins.emailext.plugins.recipients.CulpritsRecipientProvider":
+                            send_to.append("culprits")
+                        elif provider.tag == \
+                                "hudson.plugins.emailext.plugins.recipients.ListRecipientProvider":
+                            send_to.append("recipients")
+                        elif provider.tag == \
+                                "hudson.plugins.emailext.plugins.recipients.FailingTestSuspectsRecipientProvider":
+                            send_to.append("failing-test-suspects-recipients")
+                        elif provider.tag == \
+                                "hudson.plugins.emailext.plugins.recipients.FirstFailingBuildSuspectsRecipientProvider":
+                            send_to.append("first-failing-build-suspects-recipients")
+            ext_email["send-to"] = send_to
+        elif email.tag == "failureCount":
+            pass  # Handled in JJB.
+        else:
+            raise NotImplementedError("tag %s of trigger is not implemented." % email.tag)
+
+    return ext_email
+
+
+def __compare_triggers(trigger, ext_email):
+    is_equal = True
+    provider_dict = {
```

Comment:
With this approach you need to provide the mapping twice (in `__process_email()` and here).

Maybe you could just change `__compare_triggers()` to be run before any `__process_email()` to just compare tags without mapping them to JJB domain and if equal the run the actual processing?

Reply:
I couldn't understand what you have meant, but I understand that I should not provide the mapping twice, therefore I have updated the code to prevent that.